core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/SystemSettingsCredentialsProvider.java

```diff
@@ -43,12 +46,28 @@
  */
 @SdkInternalApi
 public abstract class SystemSettingsCredentialsProvider implements AwsCredentialsProvider {
+
+    private AwsCredentialsProvider credentialsProvider;
+
     @Override
     public AwsCredentials resolveCredentials() {
+
+        if (credentialsProvider == null) {
+            resolveProvider();
+        }
+
+        return credentialsProvider.resolveCredentials();
+    }
+
+    private void resolveProvider() {
         String accessKey = trim(loadSetting(SdkSystemSetting.AWS_ACCESS_KEY_ID).orElse(null));
         String secretKey = trim(loadSetting(SdkSystemSetting.AWS_SECRET_ACCESS_KEY).orElse(null));
         String sessionToken = trim(loadSetting(SdkSystemSetting.AWS_SESSION_TOKEN).orElse(null));
 
+        String roleArn = trim(loadSetting(SdkSystemSetting.AWS_ROLE_ARN).orElse(null));
```

Comment:
Should this also be used to assume a role using the assume-role credentials provider?

Reply:
This was brought up on the Go PR as well. I think it is definitely something that we should support but might do it as a separate PR.

Reply:
It may not be backwards-compatible to add it later, so should we figure it out now?

Reply:
It should be fine to add it backwards compatibly as it's presence without a web identity token would mean use regular assume role. I wouldn't want to add it before working through the process to add to all SDKs since none support it today AFAIK.

Reply:
If they specify just role_arn with this change, it'll use regular credentials. Later, it would start using assume role credentials. That seems backwards-incompatible to me.